sdk/devcenter/developer-devcenter-rest/test/public/devBoxTests.spec.ts

```diff
@@ -0,0 +1,541 @@
+    }
+
+    expect(devBoxes.length).to.equal(1);
+    expect(devBoxes[0].name).to.equal(devboxName);
+
+    await deleteDevBox();
+  });
+
+  it("GetAllDevBoxesByUser", async function () {
+    await createDevBox();
+
+    const devboxesListResponse = await client.path("/users/{userId}/devboxes", userId).get();
+
+    if (isUnexpected(devboxesListResponse)) {
+      throw devboxesListResponse.body.error;
+    }
+
+    const devBoxes: DevBoxOutput[] = [];
+    console.log("Iterating through dev boxes results:");
+
+    for await (const devBox of paginate(client, devboxesListResponse)) {
+      const { name } = devBox;
+      console.log(`Received DevBox "${name}"`);
+      devBoxes.push(devBox);
+    }
+
+    expect(devBoxes.length).to.equal(1);
+    expect(devBoxes[0].name).to.equal(devboxName);
+
+    await deleteDevBox();
+  });
+
+  it("GetDevBoxes", async function () {
+    await createDevBox();
+
+    const devboxesListResponse = await client.path("/users/{userId}/devboxes", userId).get();
+
+    if (isUnexpected(devboxesListResponse)) {
+      throw devboxesListResponse.body.error;
+    }
+
+    const devBoxes: DevBoxOutput[] = [];
+    console.log("Iterating through dev boxes results:");
+
+    for await (const devBox of paginate(client, devboxesListResponse)) {
+      const { name } = devBox;
+      console.log(`Received DevBox "${name}"`);
+      devBoxes.push(devBox);
+    }
+
+    expect(devBoxes.length).to.equal(1);
+    expect(devBoxes[0].name).to.equal(devboxName);
+
+    await deleteDevBox();
+  });
+
+  it("GetActions", async function () {
+    await createDevBox();
+
+    const actionsListResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions",
+        projectName,
+        userId,
+        devboxName
+      )
+      .get();
+
+    if (isUnexpected(actionsListResponse)) {
+      throw actionsListResponse.body.error;
+    }
+
+    const actions: DevBoxActionOutput[] = [];
+    console.log("Iterating through dev box actions results:");
+
+    for await (const action of paginate(client, actionsListResponse)) {
+      const { name } = action;
+      console.log(`Received DevBox action"${name}"`);
+      actions.push(action);
+    }
+
+    expect(actions.length).to.equal(1);
+    expect(actions[0].name).to.equal("schedule-default");
+
+    await deleteDevBox();
+  });
+
+  it("GetAction", async function () {
+    await createDevBox();
+
+    const actionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions/{actionName}",
+        projectName,
+        userId,
+        devboxName,
+        "schedule-default"
+      )
+      .get();
+
+    if (isUnexpected(actionResponse)) {
+      throw actionResponse.body.error;
+    }
+
+    expect(actionResponse.body.name).to.equal("schedule-default");
+
+    await deleteDevBox();
+  });
+
+  it("SkipAction", async function () {
+    await createDevBox();
```

Comment:
Sanity checking, this _does_ create a dev box that has a schedule applied to it, right?

Reply:
This call simply creates a devbox, and the schedule is a resource at the pool level. Therefore, it would be applied during the dev-box creation process. The session records for these tests are generated against the pool with pre-configured schedules.